applications/storage_settings/scenes/storage_settings_scene_sd_info.c

```diff
@@ -24,13 +24,44 @@ void storage_settings_scene_sd_info_on_enter(void* context) {
             dialog_ex, "Try to reinsert\nor format SD\ncard.", 3, 19, AlignLeft, AlignTop);
         dialog_ex_set_center_button_text(dialog_ex, "Ok");
     } else {
+        char unit_kb[] = "KB";
+        char unit_mb[] = "MB";
+        char unit_gb[] = "GB";
+
+        double sd_total_val = (double)sd_info.kb_total;
+        char* sd_total_unit = unit_kb;
+        double sd_free_val = (double)sd_info.kb_free;
+        char* sd_free_unit = unit_kb;
+
+        if(sd_total_val > 1024) {
```

Comment:
Better solution will be to implement formatting function that takes double* and const char**